hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/math/HazelcastDoubleBiFunction.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.operators.math;
+
+import com.hazelcast.sql.impl.calcite.validate.HazelcastCallBinding;
+import com.hazelcast.sql.impl.calcite.validate.operand.TypedOperandChecker;
+import com.hazelcast.sql.impl.calcite.validate.operators.ReplaceUnknownOperandTypeInference;
+import com.hazelcast.sql.impl.calcite.validate.operators.common.HazelcastFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+
+import static org.apache.calcite.sql.type.SqlTypeName.BIGINT;
+
+/**
+ * Function that accepts two DOUBLE arguments and produces a DOUBLE result.
+ */
+public class HazelcastDoubleBiFunction extends HazelcastFunction {
+    public HazelcastDoubleBiFunction(String name) {
+        super(
+            name,
+            SqlKind.OTHER_FUNCTION,
+            ReturnTypes.DOUBLE_NULLABLE,
+            new ReplaceUnknownOperandTypeInference(BIGINT),
```

Comment:
Shouldn't this be `DOUBLE`? Seems that we don't test with a parameter that contains some decimal places.

Reply:
You are correct. I wonder why `ReplaceUnknownOperandTypeInference` creates with `BIGINT` argument in [HazelcastDoubleFunction](https://github.com/hazelcast/hazelcast/blob/40d555dc1110da9f63fd66f504c83d4382d0384b/hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/math/HazelcastDoubleFunction.java#L40).
I still don't know our/Calcite's type inference algorithm well, so, maybe @devozerov has thoughts about it?

Reply:
I guess that's a mistake too. But I debugged it and it seems that it doesn't matter much: the validated node type is overwritten by the `TypedOperandChecker` that forces a DOUBLE on the dynamic parameter.